.github/workflows/test-action.yml

```diff
@@ -0,0 +1,16 @@
+name: Test Action
+on: push
+
+jobs: 
+  test-job:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with: 
+          ref: 'actions-implementation'
```

Comment:
Nope. That prevents this from running on any other branch than this one, remove that qualifier. GA is pretty smart and knows what you mean when you ask it to checkout the repo